sass/partials/layout/_constrain.scss

```diff
@@ -1,15 +1,25 @@
 // @file
 // Styles for constraining the page width.
 
-.l-constrain {
+@mixin constrain($width: em($max-page-width), $padding: true) {
   margin-left: auto;
   margin-right: auto;
-  max-width: em($max-page-width);
-  padding-left: $gutter-width / 2;
-  padding-right: $gutter-width / 2;
+  max-width: $width;
+
+  @if $padding {
+    padding-left: $gutter-width / 2;
+    padding-right: $gutter-width / 2;
+  }
+}
+
+.l-constrain {
+  @include constrain;
+}
+
+.l-constrain--global {
+  @include constrain(em(2500px), false);
 }
 
 .l-constrain--no-padding {
-  padding-left: 0;
-  padding-right: 0;
+  @include constrain(em($max-page-width), false);
```

Comment:
I think this variant should only have the padding-left/right properties since that’s the only thing this variant overrides.

The only reason to add all of the other styles would be to allow people to only add the l-constrain--no-padding class to an element instead of class="l-constrain l-constrain--no-padding".